e2e/cypress/integration/messaging/permalink_message_edit_spec.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// [number] indicates a test step (e.g. # Go to a page)
+// [*] indicates an assertion (e.g. * Check the title)
+// Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Permalink message edit', () => {
+    it('M18717 - Edit a message in permalink view', () => {
+        // # Login as "user-1" and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+
+        const searchWord = 'searchtest';
+
+        // # Post message
+        cy.postMessage(searchWord);
+
+        // # Search for searchWord
+        cy.get('#searchBox').type(searchWord).type('{enter}').then(() => {
+            // # Jump to permalink view
+            cy.get('.search-item__jump').first().click();
+
+            cy.getLastPostId().then((postId) => {
+                // # Click on ... button of last post matching the searchWord
+                cy.clickPostDotMenu(postId);
+
+                // # Click on edit post
+                cy.get(`#edit_post_${postId}`).click();
+
+                // # Add new text in edit box
+                cy.get('#edit_textbox').clear().type('edited searchtest');
```

Comment:
Please add const for edited text so you may re-use as needed: ``const editedText = `edited - ${searchWord}`;``